gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -321,100 +357,79 @@ MM_MemoryPoolLargeObjects::calculateTargetLOARatio(MM_EnvironmentBase* env, uint
 	OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
 	bool debug = _extensions->debugLOAResize;
 	double newLOARatio = _currentLOARatio;
-
-	/* Calculate what ratio of LOA is free */
+	float maxLOAFreeRatio = ((float)_extensions->heapFreeMaximumRatioMultiplier) / ((float)_extensions->heapFreeMinimumRatioDivisor);
 	uintptr_t loaFreeBytes = _memoryPoolLargeObjects->getActualFreeMemorySize();
-	double loaRatioFree;
 
+	//shift elements to make room for current loa free Ratio
+	for (int i = _extensions->loaFreeHistorySize; i > 0 ; i--){
+		_loaFreeRatioHistory[i] = _loaFreeRatioHistory[i-1];
+	}
 	if (0 == _loaSize) {
-		loaRatioFree = (double)0;
+		_loaFreeRatioHistory[0] = (double)0;
```

Comment:
I added that comment to where I initialize the array to all 0's to start in the initialize function. Here I'm setting it to zero to avoid division by 0 if loaSize is 0. Is that okay?